user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -500,36 +516,65 @@ define([
 
     updateData: function () {
       var myself = this;
+
+      const trashFolder = {
+        "file": {
+          "trash": "trash",
+          "createdDate": "1365427106132",
+          "fileSize": "0",
+          "folder": true,
+          "hidden": "false",
+          "objectId:": jQuery.i18n.prop('trash'),
+          "locale": "en",
+          "locked": "false",
+          "name": jQuery.i18n.prop('trash'),
+          "ownerType": "-1",
+          "path": ".trash",
+          "title": jQuery.i18n.prop('trash'),
+          "versioned": "false"
+        }
+      };
+
       myself.set("runSpinner", true);
-      myself.fetchData("/", function (response) {
-        var trash = {
-          "file": {
-            "trash": "trash",
-            "createdDate": "1365427106132",
-            "fileSize": "0",
-            "folder": "true",
-            "hidden": "false",
-            "id:": jQuery.i18n.prop('trash'),
-            "locale": "en",
-            "locked": "false",
-            "name": jQuery.i18n.prop('trash'),
-            "ownerType": "-1",
-            "path": ".trash",
-            "title": jQuery.i18n.prop('trash'),
-            "versioned": "false"
+      myself.clearTreeCache();
+      myself.fetchTreeRootData(function (response) {
+        //Add trash folder once to the first Repository folder
+        //This should only ever happen once anyway, but implemented safely
+        for( let i=0; i < response.children.length; i++){
+          let childFolder = response.children[0];
```

Comment:
Should this be `children[i]`?

Reply:
oops, yes! It was working because we happened to have `Repository` as our first folder... I probably over-future-proofed this block... 😅